supervisor/iptablesctrl/update_portsets.go

```
@@ -0,0 +1,126 @@
+package iptablesctrl
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/aporeto-inc/trireme/cache"
+	"github.com/bvandewalle/go-ipset/ipset"
+)
+
+const (
+	procNetTCPFile                    = "/proc/net/tcp"
+	portSetUpdateIntervalMilliseconds = 500
+	portEntryTimeout                  = 60
+	uidFieldOffset                    = 7
+	procHeaderLineNum                 = 1
+	portOffset                        = 1
+	ipPortOffset                      = 1
+	sockStateOffset                   = 3
+	sockListeningState                = "0A"
+	hexFormat                         = 16
+	integerSize                       = 64
+)
+
+func getUIDPortSetMappings(i *Instance, uid string) (interface{}, error) {
+	return i.UIDToPortSet.Get(uid)
+}
+
+func getUIDPortMappings(i *Instance, uid string) (interface{}, error) {
+	return i.UIDToPorts.Get(uid)
+}
+
+// InitPortSetTask This go routine is called on supervisor start
+func InitPortSetTask(i *Instance) {
+	t := time.NewTicker(portSetUpdateIntervalMilliseconds * time.Millisecond)
+	for range t.C {
+		// Update PortSet periodically.
+		updatePortSets(i)
+	}
+}
+
+func updatePortSets(i *Instance) {
```

This method should return an error